authentication/src/main/kotlin/com/uber/sdk2/auth/AuthActivity.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2024. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.sdk2.auth
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.text.TextUtils
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.lifecycleScope
+import com.uber.sdk2.auth.api.AuthProviding
+import com.uber.sdk2.auth.api.exception.AuthException
+import com.uber.sdk2.auth.api.exception.AuthException.Companion.AUTH_CODE_INVALID
+import com.uber.sdk2.auth.api.request.AuthContext
+import com.uber.sdk2.auth.api.response.AuthResult
+import com.uber.sdk2.auth.internal.AuthProvider
+import com.uber.sdk2.core.utils.CustomTabsHelper
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class AuthActivity : AppCompatActivity() {
+
+  private var authProvider: AuthProviding? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    val authContext =
+      intent.getParcelableExtra<AuthContext>(AUTH_CONTEXT)
+        ?: throw IllegalStateException("AuthContext is required")
+
+    authProvider = AuthProvider(this, authContext)
+    init()
+  }
+
+  private fun init() {
+    authProvider?.let {
+      lifecycleScope.launch(Dispatchers.IO) {
+        val authResult = it.authenticate()
+        withContext(Dispatchers.Main) {
```

Comment:
why dont we use `withContext()` in `authenticate()` instead?

Method should be safe to call from any context.